mlir/docs/Tutorials/Toy/Ch-4.md

```diff
@@ -189,6 +189,12 @@ CallInterfaceCallable GenericCallOp::getCallableForCallee() {
   return getAttrOfType<SymbolRefAttr>("callee");
 }
 
+/// Set the callee for the generic call operation, this is required by the call
+/// interface.
+void GenericCallOp::setCalleeFromCallable(CallInterfaceCallable callee) {
+  (*this)->setAttr("callee", dyn_cast<SymbolRefAttr>(callee));
+}
+
```

Comment:
Does a cast from `CallInterfaceCallable` to an attribute work here?

Reply:
Doesn't work with error:
```
mlir/include/mlir/IR/StorageUniquerSupport.h:111:5: error: static_assert failed due to requirement 'std::is_convertible<mlir::SymbolRefAttr, const mlir::CallInterfaceCallable *>::value' "casting from a non-convertible type"
    static_assert(std::is_convertible<ConcreteT, T>::value,
    ^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
Not at all. I was just wondering as I didn't remember CallInterfaceCallable definition.